public/app/features/plugins/extensions/utils.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+
+import { type PluginExtensionLinkConfig, type PluginExtensionEventHelpers } from '@grafana/data';
+import { Modal } from '@grafana/ui';
+import appEvents from 'app/core/app_events';
+import { ShowModalReactEvent } from 'app/types/events';
+
+export function logWarning(message: string) {
+  console.warn(`[Plugin Extensions] ${message}`);
+}
+
+export function isPluginExtensionLinkConfig(
+  extension: Pick<PluginExtensionLinkConfig, 'title' | 'description'> | undefined
```

Comment:
It might be worth to introduce a type for the `Pick<PluginExtensionLinkConfig, 'title' | 'description'> | undefined` since we have defined it in at least two different places and they already have diverged.